Sources/Xcodeproj/generate().swift

```diff
@@ -223,3 +231,41 @@ func generateSchemes(
         ).generate()
     }
 }
+
+// Find and return non-source files in the source directories and root that should be added
+// as a reference to the project.
+func getExtraFilesFor(package: ResolvedPackage, in workingCheckout: WorkingCheckout? = nil) throws -> [AbsolutePath] {
+    let srcroot = package.path
+    var extraFiles = try findNonSourceFiles(path: srcroot)
+
+    for target in package.targets {
+        let sourcesDirectory = target.sources.root
+        if localFileSystem.isDirectory(sourcesDirectory) {
+            let sourcesExtraFiles = try findNonSourceFiles(path: sourcesDirectory, recursively: true)
+            extraFiles.append(contentsOf: sourcesExtraFiles)
+        }
+    }
+
+    if let checkout = workingCheckout {
```

Comment:
Hmm, maybe we shouldn't do this unless we have a working checkout because otherwise we will end up adding dot files like .DS_Store

Reply:
Yeah, I can agree on this, also I think in most cases there will be a working checkout, so requiring it for this feature shouldn't cause many issues.